### PR TITLE
Fixed the issue that the configured out of order sealerList caused inconsistent genesis hashes

### PR DESCRIPTION
**Pull request by chuwen95**
fix  issue #2213

### CODE REVIEW

File: libinitializer/LedgerInitializer.h [hunk 1]
```diff
@@ -63,13 +64,13 @@ class LedgerInitializer : public std::enable_shared_from_this<LedgerInitializer>
 
     void startAll()
     {
-        if (m_ledgerManager)
+        if (nullptr != m_ledgerManager)
```
Comment (cyjseagull) [inline libinitializer/LedgerInitializer.h:67]: No need to change here.

File: libinitializer/LedgerInitializer.h [hunk 2]
```diff
@@ -63,13 +64,13 @@ class LedgerInitializer : public std::enable_shared_from_this<LedgerInitializer>
...
             m_ledgerManager->startAll();
     }
 
     void stopAll()
     {
-        if (m_ledgerManager)
+        if (nullptr != m_ledgerManager)
```
Comment (cyjseagull) [inline libinitializer/LedgerInitializer.h:73]: No need to change here.

File: libledger/LedgerParam.cpp [hunk 3]
```diff
@@ -427,6 +427,16 @@ void LedgerParam::initConsensusConfig(ptree const& pt)
     // if the consensus node id is invalid, throw InvalidConfiguration exception
     parsePublicKeyListOfSection(mutableConsensusParam().sealerList, pt, "consensus", "node.");
     std::stringstream nodeListMark;
+    // sort the sealerList
+    if (g_BCOSConfig.version() >= V2_9_0)
```
Comment (cyjseagull) [inline libledger/LedgerParam.cpp:431]: Maybe no need use `lessThan`.

use `std::less<h512>` or use `std::sort(mutableConsensusParam().sealerList.begin(),
            mutableConsensusParam().sealerList.end())` only.

File: libledger/LedgerParam.cpp [hunk 4]
```diff
@@ -427,6 +427,16 @@ void LedgerParam::initConsensusConfig(ptree const& pt)
     // if the consensus node id is invalid, throw InvalidConfiguration exception
     parsePublicKeyListOfSection(mutableConsensusParam().sealerList, pt, "consensus", "node.");
     std::stringstream nodeListMark;
+    // sort the sealerList
+    if (g_BCOSConfig.version() >= V2_9_0)
+    {
+        const auto lessThan = [](const h512& a, const h512& b) -> bool {
+            return (a < b) ? true : false;
+        };
+        std::sort(mutableConsensusParam().sealerList.begin(),
```
Comment (ywy2090) [inline libledger/LedgerParam.cpp:436]: std::sort should replace by std::stable_sort

File: libledger/LedgerParam.cpp (hunk 3 above)
Comment (chuwen95) [reply libledger/LedgerParam.cpp:431]: yes, you are right, thanks for the suggestion!

Comment (chuwen95) [reply libledger/LedgerParam.cpp:431]: > Maybe no need use `lessThan`.
> 
> use `std::less<h512>` or use `std::sort(mutableConsensusParam().sealerList.begin(), mutableConsensusParam().sealerList.end())` only.

yes, you are right, thanks for the suggestion!

**Review by ywy2090 — APPROVED** [review]

**Pull request merged**